packages/account/src/Sections/Profile/PhoneVerification/__test__/otp-verification.spec.tsx

```diff
@@ -52,15 +61,82 @@ describe('ConfirmPhoneNumber', () => {
 
     it('should render Verify your number in OTP Verification', () => {
         store.ui.should_show_phone_number_otp = true;
+        (useSendOTPVerificationCode as jest.Mock).mockReturnValue({
+            sendPhoneOTPVerification: jest.fn(),
+        });
         renderComponent();
         expect(screen.getByText(/Verify your number/)).toBeInTheDocument();
         expect(screen.getByText(/Enter the 6-digit code sent to you via SMS at :/)).toBeInTheDocument();
     });
 
     it('should render whatsapp when phone_verification_type is whatsapp', () => {
         store.ui.should_show_phone_number_otp = true;
+        (useSendOTPVerificationCode as jest.Mock).mockReturnValue({
+            sendPhoneOTPVerification: jest.fn(),
+        });
         phone_verification_type = 'whatsapp';
         renderComponent();
         expect(screen.getByText(/WhatsApp/)).toBeInTheDocument();
     });
+
+    it('should contain value of 123456 for otp textfield component', () => {
+        store.ui.should_show_phone_number_otp = true;
+        (useSendOTPVerificationCode as jest.Mock).mockReturnValue({
+            sendPhoneOTPVerification: jest.fn(),
+            setPhoneOtpErrorMessage: jest.fn(),
+        });
+        renderComponent();
+        const otp_textfield = screen.getByRole('textbox');
+        userEvent.type(otp_textfield, '123456');
+        expect(otp_textfield).toHaveValue('123456');
+    });
+
+    it('should render mockSendPhoneOTPVerification when Verify button is clicked', () => {
+        store.ui.should_show_phone_number_otp = true;
+        (useSendOTPVerificationCode as jest.Mock).mockReturnValue({
+            sendPhoneOTPVerification: mockSendPhoneOTPVerification,
+            setPhoneOtpErrorMessage: jest.fn(),
+        });
+        renderComponent();
+        const otp_textfield = screen.getByRole('textbox');
+        const verify_button = screen.getByRole('button', { name: 'Verify' });
+        userEvent.type(otp_textfield, '123456');
+        expect(verify_button).toBeEnabled();
+        userEvent.click(verify_button);
+        expect(mockSendPhoneOTPVerification).toBeCalledTimes(1);
+    });
+
+    it('should show error message when API returns error', () => {
+        store.ui.should_show_phone_number_otp = true;
+        (useSendOTPVerificationCode as jest.Mock).mockReturnValue({
+            sendPhoneOTPVerification: mockSendPhoneOTPVerification,
+            phone_otp_error_message: 'Error Message',
+        });
+        renderComponent();
+        expect(screen.getByText(/Error Message/)).toBeInTheDocument();
```

Comment:
are we showing error messages from API directly or do we use any error codes to map the error messages?

Reply:
We use error codes to map the error messages 🙏